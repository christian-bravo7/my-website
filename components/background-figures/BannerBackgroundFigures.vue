<template>
  <div class="hidden md:block absolute overflow-hidden w-full h-full pointer-events-none">
    <span class="text-pink-500 dark:text-blue-500 text-opacity-30 dark:text-opacity-30">
      <CircleIcon
        class="absolute w-6 h-6 movable-figure movable-figure--1 fill-current"
        :class="currentRoute"
      />
    </span>
    <span class="text-pink-500 dark:text-blue-500 text-opacity-30 dark:text-opacity-30">
      <CircleIcon
        class="absolute w-7 h-7 movable-figure movable-figure--2 fill-current"
        :class="currentRoute"
      />
    </span>
    <span class="text-pink-500 dark:text-blue-500 text-opacity-30 dark:text-opacity-30">
      <CircleIcon
        class="absolute w-8 h-8 movable-figure movable-figure--3 fill-current"
        :class="currentRoute"
      />
    </span>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';

import CircleIcon from '~/assets/img/circle-outline.svg?inline';

@Component({
  components: {
    CircleIcon,
  },
})
export default class BannerBackgroundFigures extends Vue {
  @Prop({ type: String, required: true })
  currentRoute!: string;
}
</script>

<style lang="scss" scoped>
.movable-figure {
  $self: &;

  transition: color 200ms, top 6s, left 6s, transform 6s;

  &.index {
    &#{$self}--1 {
      top: 30%;
      left: 92%;
    }
    &#{$self}--2 {
      top: 20%;
      left: 10%;
    }
    &#{$self}--3 {
      top: 80%;
      left: 60%;
    }
  }

  &.me {
    &#{$self}--1 {
      top: 20%;
      left: 80%;
    }
    &#{$self}--2 {
      top: 50%;
      left: 10%;
    }
    &#{$self}--3 {
      top: 90%;
      left: 70%;
    }
  }

  &.blog {
    &#{$self}--1 {
      top: 10%;
      left: 88%;
    }
    &#{$self}--2 {
      top: 60%;
      left: 5%;
    }
    &#{$self}--3 {
      top: 94%;
      left: 50%;
    }
  }
}

</style>
