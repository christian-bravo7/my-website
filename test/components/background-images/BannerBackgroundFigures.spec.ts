import { mount } from '@vue/test-utils';

import BannerBackgroundFigures from '@/components/background-figures/BannerBackgroundFigures.vue';

describe('BannerBackgroundFigures component', () => {
  it('should render correctly', () => {
    const wrapper = mount(BannerBackgroundFigures, {
      propsData: {
        currentRoute: 'about',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should set class name equals to currentRoute', () => {
    const wrapper = mount(BannerBackgroundFigures, {
      propsData: {
        currentRoute: 'home',
      },
    });

    const figures = wrapper.findAll('figure.absolute');
    const figuresWithClass = figures.filter(wrapper => wrapper.classes('home'));

    expect(figures.length).toBe(figuresWithClass.length);
  });
});
