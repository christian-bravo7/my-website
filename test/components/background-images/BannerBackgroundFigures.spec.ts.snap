// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerBackgroundFigures component should render correctly 1`] = `<div class="hidden md:block absolute overflow-hidden w-full h-full pointer-events-none" currentroute="about"><span class="text-pink-500 dark:text-blue-500 text-opacity-30 dark:text-opacity-30"><svg class="absolute w-6 h-6 movable-figure movable-figure--1 fill-current /"></svg></span> <span class="text-pink-500 dark:text-blue-500 text-opacity-30 dark:text-opacity-30"><svg class="absolute w-7 h-7 movable-figure movable-figure--2 fill-current /"></svg></span> <span class="text-pink-500 dark:text-blue-500 text-opacity-30 dark:text-opacity-30"><svg class="absolute w-8 h-8 movable-figure movable-figure--3 fill-current /"></svg></span></div>`;
